guides/common/modules/con_transport-modes-for-remote-execution.adoc

```diff
@@ -0,0 +1,32 @@
+[id="transport-modes-for-remote-execution_{context}"]
+= Transport Modes for Remote Execution
+
+You can configure your {Project} to use two different modes of transport for remote job execution.
+
+By default, remote execution transports job details over SSH.
+This is referred to as _push mode_.
+The SSH service must be enabled and active on the target hosts.
+The remote execution {SmartProxy} must have access to the SSH port on the target hosts.
+Unless you have a different setting, the standard SSH port is 22.
+
+If your infrastructure prohibits connections from {SmartProxy} to the target hosts, you can use _pull mode_.
+When using the pull mode, the hosts initiate all connections and there are no outgoing connections from {SmartProxy}.
+{SmartProxy} notifies the host of a new job over Message Queueing Telemetry Transport (MQTT).
+The target host subscribes to the MQTT broker on {SmartProxy} for job notifications.
+After the host receives a notification, it pulls job details from {SmartProxy} over HTTPS, runs the job, and reports results back to {SmartProxy}.
+
+NOTE: The pull mode is only applicable to Script provider jobs.
+
+To use the pull mode, you must enable it on {SmartProxyServer} and on the target hosts.
+
+.Additional resources
+* xref:enabling-pull-mode-on-proxy_{context}[]
+ifdef::katello,orcharhino,satellite[]
+* To enable the pull mode on an existing host, continue with xref:migrating-a-host-to-use-pull-mode_{context}[].
+endif::[]
+ifdef::managing-hosts[]
+* To enable the pull mode on a new host, continue either with xref:Creating_a_Host_{context}[] or xref:Registering_Hosts_{context}[].
+endif::[]
+ifndef::managing-hosts[]
+* To enable the pull mode on a new host, continue either with {ManagingHostsDocURL}Creating_a_Host_managing-hosts[Creating a Host] or {ManagingHostsDocURL}Registering_Hosts_managing-hosts[Registering Hosts] in _{ManagingHostsDocTitle}_.
```

Comment:
This section is useful to add.  (The rest of this file is covered by https://github.com/theforeman/foreman-documentation/pull/1548.)